### ISSUE TITLE
Add an ability to assign custom highlight without writing an appmodule.

### ISSUE THREAD

**Issue by lukaszgo1** [nvaccess/nvda#9035]
### Is your feature request related to a problem? Please describe.
In many edit boxes in which NVDA is using display model to access innformations the selection isn't announced. Examples of those include #4535 #8946 and #4756.
In most of this cases NVDA recognizes the fact, that the selected text has different color,  so fixing it is only a matter of telling, that this color is the highlight.
### Describe the solution you'd like
1. For the app in question create a configuration profile.
2. select text.
3. move to the one of selected character with the object review.
4. press some defined keystroke and verify that color is picked up correctly.
5. Write the color to the config profile and use it when selecting text in this program.

### Describe alternatives you've considered
I do not know if it can be retrieved from the config profile. If it cannot be perchaps  an easy way to provide this info in the app module easy enough to be doable even for non programmers.
### Additional context
It is de facto standard function in the other commercial screen readers which is able to solve many accessibility problems in a really short time and without programming experience.

**Comment by leonardder** [comment]
I think configuration profiles aren't suitable for this. Fixing this in an appModule shouldn't be very difficult, and #10040.

> ... which is able to solve many accessibility problems in a really short time and without programming experience.

Yes, I have been able to solve many accessibility problems with this approach. I once filed #8414 for this, but it turned out to be too big. I'm now intending to split that pr up into smaller ones. The first one is #10040, but having a SelectableText class is another requirement to solve many a11y problems out of the box.

**Comment by lukaszgo1** [comment]
@leonardder wrote:

> 
> 
> I think configuration profiles aren't suitable for this. Fixing this in an appModule shouldn't be very difficult, and #10040.
> 
> > ... which is able to solve many accessibility problems in a really short time and without programming experience.


I am afraid that this still wouldn't help much for an non programmer. In a commercial screen readers even very uninitiated user is able to assign custom highlight as long as he understood the concept. Also highlight color may differ between systems, so it would be beneficial for users to allow them to customize it to their systems. Out of interest how are you solving such problems at the moment?

**Comment by leonardder** [comment]
> Out of interest how are you solving such problems at the moment?

All with custom appModules, but the code needed inside such an appModule is pretty small when #10040 and #10058 would be accepted. I think first steps would be offering more in core abilities to solve both easy and complex issues. I'm using code from mentioned pull requests in like 70% of appModules written for legacy software in enterprise environments.

The problem with having an ability to save this information in other places is that it is hard to come up with a GUI that covers everything.

See also https://github.com/babbagecom/objenhancer
